<h4>Railway Cars and Eigenvalue Method</h4>

In the three-railway-car system pictured below, suppose that
cars 1 and 3 each weigh 32 tons, that car 2 weighs 8 tons,
and that each spring constant is 4 tons/ft. If $x_1'(0)=v_0$
and $x_2'(0)=x_3'(0)=0$, show that the two springs are compressed 
until $t=\pi/2$ and that
$$
x_1'(t)=-\frac{1}{9}v_0\mbox{ and }x_2'(t)=x_3'(t)=\frac{8}{9}v_0
$$
thereafter. Thus car 1 rebounds, but cars 2 and 3 continue
with the same velocity.

<img src="./diagrams/wwulnbvt.png" width="70%" height="70%">

In [12]:
using LinearAlgebra

In [13]:
M=[32 0 0
0 8 0
0 0 32]

3×3 Matrix{Int64}:
 32  0   0
  0  8   0
  0  0  32

In [14]:
K=[-4 4 0
4 -8 4
0 4 -4]

3×3 Matrix{Int64}:
 -4   4   0
  4  -8   4
  0   4  -4

In [15]:
A=inv(M)*K

3×3 Matrix{Float64}:
 -0.125   0.125   0.0
  0.5    -1.0     0.5
  0.0     0.125  -0.125

In [16]:
eigvals(A)

3-element Vector{Float64}:
 -1.125
 -0.12499999999999997
  1.9285316915676624e-17

In [18]:
ev=eigvecs(A)

3×3 Matrix{Float64}:
  0.123091   0.707107     0.57735
 -0.984732   2.52792e-17  0.57735
  0.123091  -0.707107     0.57735

In [21]:
ev[:,1]*(1/ev[1,1]) # eigenvector 1

3-element Vector{Float64}:
  1.0
 -7.999999999999988
  0.999999999999998

In [22]:
ev[:,2]*(1/ev[1,2]) # eigenvector 2, eigenvector 3 is obvious

3-element Vector{Float64}:
  1.0
  3.575020688115945e-17
 -1.0000000000000004

In [23]:
M1=[1 1 1
1 -8 0
1 1 -1]

3×3 Matrix{Int64}:
 1   1   1
 1  -8   0
 1   1  -1

In [25]:
det(M1) # therefore a_i=0

18.0

In [26]:
using SymPy

In [27]:
@vars w # this is v_0

(w,)

In [28]:
M2=[1 3/sqrt(8) 1/sqrt(8)
1 -8*3/sqrt(8) 0
1 3/sqrt(8) -1/sqrt(8)]

3×3 Matrix{Float64}:
 1.0   1.06066   0.353553
 1.0  -8.48528   0.0
 1.0   1.06066  -0.353553

In [33]:
v2=[1;0;0] # this should be [w;0;0] for symbolic calculations -- assuming v_0=1

3-element Vector{Int64}:
 1
 0
 0

In [34]:
B=inv(M2)*v2

3-element Vector{Float64}:
 0.4444444444444444
 0.05237828008789242
 1.4142135623730951

In [35]:
b1=B[1,1]

0.4444444444444444

In [36]:
b2=B[2,1]

0.05237828008789242

In [37]:
b3=B[3,1]

1.4142135623730951

In [38]:
x1p(t)=b1+(3/sqrt(8))*b2*cos((3/sqrt(8))*t)+(1/sqrt(8))*b3*cos((1/sqrt(8))*t)

x1p (generic function with 1 method)

In [39]:
x3minusx2(t)=9*b2*sin((3/sqrt(8))*t)-b3*sin((1/sqrt(8))*t)

x3minusx2 (generic function with 1 method)

In [41]:
using Roots

In [43]:
z=find_zero(x3minusx2,8)

8.885726750148962

In [44]:
x1p(z) # velocity of car 1 at disengagement

-0.11111111101543258

In [45]:
x2p(t)=b1+((-8)*3/sqrt(8))*b2*cos((3/sqrt(8))*t)

x2p (generic function with 1 method)

In [46]:
x2p(z)

0.8888888885061748

In [47]:
x3p(t)=b1+(3/sqrt(8))*b2*cos((3/sqrt(8))*t)-(1/sqrt(8))*b3*cos((1/sqrt(8))*t)

x3p (generic function with 1 method)

In [48]:
x3p(z)

0.8888888888888888

I do not get EPC's desired result $t=\pi/2$ when the cars disengage, despite checking and rechecking. My results look good on desmos, and even the velocities of the cars are correct according to my results. According to my calculations, the cars disengage after about 8.9 seconds (sounds like a long time). 